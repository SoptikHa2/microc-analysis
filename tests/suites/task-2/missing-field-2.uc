main() {
    var n, k;

    k = {a: 1};
    n = alloc {a: &k, b: 2};
    output (*(*n).a).b; // error: missing field

    return 0;
}