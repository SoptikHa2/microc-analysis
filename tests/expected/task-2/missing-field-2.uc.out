---
test: type suites/task-2/missing-field-2.uc
---
suites/task-2/missing-field-2.uc:6:15: type error: Cannot unify '◇' and '⟦(*(*n).a).b[6:21]⟧' (with canonicals '◇' and 'int')
      output (*(*n).a).b; // error: missing field
                ^


exit: 1
